addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -24,13 +32,25 @@ const GalleryWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onClickImg: function (ev) {
-        if (this.$modal || ev.currentTarget.matches("a > img")) {
+        const clickedEl = ev.currentTarget;
+        if (this.$modal || clickedEl.matches("a > img")) {
             return;
         }
         var self = this;
-        var $cur = $(ev.currentTarget);
 
-        var $images = $cur.closest('.s_image_gallery').find('img');
+        let imageEls = this.el.querySelectorAll("img");
+        const currentImageEl = clickedEl.closest("img");
+        const currentImageIndex = [...imageEls].indexOf(currentImageEl);
+        // We need to reset the images to their original source because it might
+        // have been changed by a mouse event (e.g. "hover effect" animation).
+        let i = 0;
+        imageEls = [...imageEls].map(el => {
+            const cloneEl = el.cloneNode(true);
+            cloneEl.src = this.originalSources[i];
+            i++;
+            return cloneEl;
+        });
```

Comment:
It works fine, but in practice, I am not sure to understand how it is working, can you explain it to me (and maybe add the explanation as a comment or in the commit message)  ? :slightly_smiling_face: 
Are the images `src` different in the `start` ? Because from what I see, what you do is:
- in `start`, save the images `src`
- on click, clone all images and set these saved src as their new src (how are they different than the current ones ?)
- these clones are then given to the render which uses these `src` to render the images (in the `website.gallery.slideshow` template)

Is it because the `src` changes when we hover the image and so in the `start`, since no animation could be played yet, it is still the image version without the animation that is saved ? :thinking:

Reply:
> Is it because the `src` changes when we hover the image and so in the `start`, since no animation could be played yet, it is still the image version without the animation that is saved ? 🤔

Yes that's exactly it